meteor/client/ui/Prompter/controller/midi-pedal-device.ts

```diff
@@ -0,0 +1,148 @@
+import { ControllerAbstract } from './lib'
+import { PrompterViewInner } from '../PrompterView'
+
+import webmidi, { Input, InputEventControlchange, WebMidi } from 'webmidi'
+
+const LOCALSTORAGEMODE = 'prompter-controller-arrowkeys'
+
+/**
+ * This class handles control of the prompter using
+ */
+export class MidiPedalController extends ControllerAbstract {
+	private prompterView: PrompterViewInner
+	private midiInput: Input | undefined
+
+	private readonly rangeRevMin = 0 // pedal "all back" position, the max-reverse-position
+	private readonly rangeNeutralMin = 35 // pedal "back" position where reverse-range transistions to the neutral range
+	private readonly rangeNeutralMax = 80 // pedal "front" position where scrolling starts, the 0 speed origin
+	private readonly rangeFwdMax = 127 // pedal "all front" position where scrolling is maxed out
+	private readonly speedMap = [1, 2, 3, 4, 5, 7, 9, 12, 17, 19, 30]
+	private readonly reverseSpeedMap = [10, 30, 50]
+
+	private updateSpeedHandle: number | null = null
+	private lastSpeed = 0
+	private currentPosition = 0
+
+	constructor(view: PrompterViewInner) {
+		super(view)
+		this.prompterView = view
+
+		// validate range settings
+		// they need to be in sequence, or the logic will break
+		if (this.rangeNeutralMin <= this.rangeRevMin) {
+			console.error('rangeNeutralMin must be larger to rangeRevMin. Pedal control will not initialize.')
+			return
+		}
+		if (this.rangeNeutralMax <= this.rangeNeutralMin) {
+			console.error('rangeNeutralMax must be larger to rangeNeutralMin. Pedal control will not initialize')
+			return
+		}
+		if (this.rangeFwdMax <= this.rangeNeutralMax) {
+			console.error('rangeFwdMax must be larger to rangeNeutralMax. Pedal control will not initialize')
+			return
+		}
+
+		webmidi.enable(this.setupMidiListeners.bind(this))
+	}
+
+	public destroy() {
```

Comment:
Shouldn't this run `webmidi.disable()` to clean up and free up any MIDI resources we have connected to?

Reply:
Added